src/partition/FileUtils.h

```diff
@@ -59,6 +65,35 @@ class InputFile {
     off_t _sz;
 };
 
+class InputFileArrow : public InputFile {
+public:
+    explicit InputFileArrow(boost::filesystem::path const &path, off_t blockSize);
+    ~InputFileArrow();
+
+    /// Return the size of the input file.
+    off_t size() const { return _sz; }
+
+    /// Return the path of the input file.
+    boost::filesystem::path const &path() const { return _path; }
+
+    int getBatchNumber() const;
```

Comment:
It's better to mark overrident methods with `override`.

Reply:
Actually, it needs to be declared like this:
```
virtual int getBatchNumber() const override;
```

Reply:
Done